HW1/P4/P4.txt

```diff
@@ -1 +1,26 @@
 # Your discussion here
+
+I use an adjacency list to represent the graph. It is a set of ordered pairs structured as (character x, np.array([neighbors of character x]))
+The graph diameter (10) implies that you only need to do 10 BFS iterations after the root/start node is set to distance 0. 
+This is because every node connected to the root node is at most a distance of 10 edges away from the root node.
+If a character does not have a defined distance during the search, it means no path exists between the root node and the 
+character with undefined distance.
+
+Number of Touched Nodes for each character:
+
+CAPTAIN AMERICA: 6408/6444
+MISS THING/MARY: 7/6444
+ORWELL: 9/6444
+
+Connected Components sizes of the undirected Marvel graph:
+
+[6408, 1, 7, 9, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+
+Connected Component sizes at each iteration:
+
+CAPTAIN AMERICA: [1, 1907, 6370, 6408]
+MISS THING/MARY: [1, 7]
+ORWELL: [1, 9]
+
```

Comment:
Nice answer. I like that you also had the connected components info here as well.